lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -472,15 +481,15 @@ class LLDBMemoryReader : public swift::remote::MemoryReader {
 
   void popLocalBuffer() {
     lldbassert(m_local_buffer);
-    m_local_buffer = 0;
+    m_local_buffer.reset();
     m_local_buffer_size = 0;
   }
 
 private:
   Process &m_process;
   size_t m_max_read_amount;
 
-  uint64_t m_local_buffer = 0;
+  llvm::Optional<uint64_t> m_local_buffer;
```

Comment:
Some enum values are zero sized. Because of the asserts in this function, `PushLocalBuffer()` previously should be only called if the buffer was non-zero. This means you'd have to write code like:

```
if (buffer)
  pushLocalBuffer(...);
DO STUFF
if (buffer)
  popLocalBuffer();
```

By using optional, the calling code can be made unconditional:

```
pushLocalBuffer(...);
DO STUFF
popLocalBuffer();
```

This works better with `make_scope_exit` too.